msgraph-proxy-plugins/Guidance/SdkGuidancePlugin.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Graph.DeveloperProxy.Abstractions;
+using System.Text.RegularExpressions;
+using Titanium.Web.Proxy.Http;
+
+namespace Microsoft.Graph.DeveloperProxy.Plugins.Guidance;
+
+public class SdkGuidancePlugin : IProxyPlugin {
+    private ISet<Regex>? _urlsToWatch;
+    private ILogger? _logger;
+    public string Name => nameof(SdkGuidancePlugin);
+
+    public void Register(IPluginEvents pluginEvents,
+                            IProxyContext context,
+                            ISet<Regex> urlsToWatch,
+                            IConfigurationSection? configSection = null) {
+        if (pluginEvents is null) {
+            throw new ArgumentNullException(nameof(pluginEvents));
+        }
+
+        if (context is null || context.Logger is null) {
+            throw new ArgumentException($"{nameof(context)} must not be null and must supply a non-null Logger", nameof(context));
+        }
+
+        if (urlsToWatch is null || urlsToWatch.Count == 0) {
+            throw new ArgumentException($"{nameof(urlsToWatch)} cannot be null or empty", nameof(urlsToWatch));
+        }
+
+        _urlsToWatch = urlsToWatch;
+        _logger = context.Logger;
+
+        pluginEvents.Request += OnRequest;
+    }
+
+    private void OnRequest(object? sender, ProxyRequestArgs e) {
+        Request request = e.Session.HttpClient.Request;
+        if (_urlsToWatch is not null && e.ShouldExecute(_urlsToWatch) && WarnNoSdk(request))
```

Comment:
I think the semantics of this plugin are different comparing to what we have right now. Right now, we only show the message when the request was issued without the SDK and the request failed: https://github.com/microsoftgraph/msgraph-developer-proxy/blob/33e42621b4bc15c53669fd3e1f3be887053a9793/msgraph-developer-proxy/ChaosEngine.cs#L317-L323

To keep this behavior intact, we should subscribe to the response event where we can check if the response is different than 200. If the requests succeeds, it would be odd for us to advise users to use the SDK to handle errors, while there aren't any, right? 😄

Reply:
I've moved this to be triggered by the AfterResponse event. 
The AfterResponse event is raised when calls are passed through to graph and when calls are intercepted during the BeforeRequest event.